sdk/include/opentelemetry/sdk/common/attribute_utils.h

```diff
@@ -29,6 +29,7 @@ using OwnedAttributeValue = nostd::variant<bool,
                                            uint32_t,
                                            int64_t,
                                            double,
+                                           const char *,
```

Comment:
Is this intended. As char pointer is not owned attribute?

Reply:
You are right. It should not be needed because we have a converter to translate `common::AttributeValue(const char *)` to `OwnedAttributeValue(std::string)`. Let me double-check again, and I should be able to remove that.. What I don't remember - if you are using this class for `Resrouce`? If you are, then it's still needed. Resource, however, has a separate method in constructor that allows to transform all `const char *` to `std::string`.

Reply:
Yeah, you are using it for `ResourceAttributes` here:
https://github.com/open-telemetry/opentelemetry-cpp/blob/12e56f9f4ad2e2a870482d89b999a9cf7c493e85/sdk/include/opentelemetry/sdk/resource/resource.h#L22

So if we do not add it to `OwnedAttributeValue`, then only `ResourceAttributes` scenario is affected. And you'd still hit the issue described in #733 !

However, when all exporters deal with `OwnedAttributeValue`, - I believe they should not be receiving any objects that contain `const char *` . Because in the transformer from `AttributeValue` to `OwnedAttributeValue` we now initialize the string value as `std::string` here:

https://github.com/open-telemetry/opentelemetry-cpp/blob/fb5393b1c4151ef3a9dcc059c1c8973191853dca/sdk/include/opentelemetry/sdk/common/attribute_utils.h#L80

That means to downstream exporters, they'd never normally have to deal with `const char *` much. So I would rather keep it. Or if you have alternate suggestions how to solve the `ResourceAttributes` issue in a nicer way, that'd be much appreciated?

One issue that I think we may want to consider, is if in future we allow initialization from map of `const char *` (which in my opinion, is natural and expected, and most people would like to see that) - **we could have avoided a memcpy in synchronous non-batching exporter scenario, akin to ETW - you do not have to own the attribute value when it is immediately serialized on user thread**.

Reply:
So I'm not changing it, because changing it breaks the `ResourceAttributes` scenario.

Reply:
@maxgolov  - Would you be removing `const char *` from `OwnedAttributeValue` in this PR, as we are changing `ResourceAttributes` implementation to directly use` common::AttributeValue` (#817- thanks for raising this issue) in separate PR? I understand we may have to ( temporarily ) convert resource string literals to std::string() in unit-tests which should be fine right?